File: 프로그래머스/0/181856. 배열 비교하기/배열 비교하기.js
function solution(arr1, arr2) {
    var answer = 0;
    
    if (arr1.length > arr2.length) {
        return 1
    } else if (arr1.length < arr2.length) {
        return -1
    } else {
        const sum1 = arr1.reduce((arr, cur) => arr + cur, 0)
        const sum2 = arr2.reduce((arr, cur) => arr + cur, 0)
        
        if (sum1 > sum2) {
            return 1
        } else if (sum1 < sum2) {
            return -1
        } else {
            return 0
        }
        
    }
    
    return answer;
}

File: 프로그래머스/0/181856. 배열 비교하기/배열 비교하기.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<int> arr1, vector<int> arr2) {
    int a = arr1.size();
    int b = arr2.size();
    
    int sum_a = 0;
    int sum_b = 0;
    
    for (int a1 : arr1) {
        sum_a += a1;
    }
    
    for (int a2 : arr2) {
        sum_b += a2;
    }
    
    if (a != b) {
        if (a > b) {
            return 1;
        } else {
            return -1;
        }
    } else {
        if (sum_a > sum_b) {
            return 1;
        } else if (sum_a < sum_b) {
            return -1;
        } else {
            return 0;
        }
    }
    
}